src/pubsub/ua_pubsub_keystorage.c

```diff
@@ -165,6 +165,12 @@ UA_PubSubKeyStorage_addKeyRolloverCallback(UA_PubSubManager *psm,
     if(!psm || !ks || !callback || timeToNextMs <= 0)
         return UA_STATUSCODE_BADINVALIDARGUMENT;
 
+    // First remove the existing callback
+    if(*callbackID != 0) {
```

Comment:
On the contrary. When the callback already exists, then we don't have to re-add it and can do an early exit.

Reply:
I don't agree. By not re-registering the callback, we risk missing timing adjustment.

Reply:
ok, I agree. We don't have to re-add it